lgc/include/lgc/state/AbiMetadata.h

```diff
@@ -207,11 +207,49 @@ constexpr unsigned int mmSPI_SHADER_USER_DATA_HS_0 = 0x2D0C; // For GFX9+, Used
 constexpr unsigned int mmSPI_SHADER_USER_DATA_PS_0 = 0x2C0C;
 constexpr unsigned int mmSPI_SHADER_USER_DATA_VS_0 = 0x2C4C;
 
+// The RSRC1 registers.
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_LS = 0x2D4A;
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_HS = 0x2D0A;
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_ES = 0x2CCA;
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_GS = 0x2C8A;
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_VS = 0x2C4A;
+constexpr unsigned mmSPI_SHADER_PGM_RSRC1_PS = 0x2C0A;
+
+// RSRC2 register. We only specify one, as each graphics shader stage has its RSRC2 register at the same
+// offset (-1) from its USER_DATA_*_0 register.
+constexpr unsigned mmSPI_SHADER_PGM_RSRC2_VS = 0x2C4B;
+
 // Other SPI register numbers in PAL metadata
 constexpr unsigned int mmPA_CL_CLIP_CNTL = 0xA204;
+constexpr unsigned mmVGT_SHADER_STAGES_EN = 0xA2D5;
 
 // Register bitfield layout.
 
+// General RSRC1 register, enough to get the VGPR and SGPR counts.
+union SPI_SHADER_PGM_RSRC1 {
+  struct {
+    unsigned int VGPRS : 6;
+    unsigned int SGPRS : 4;
+    unsigned int : 22;
+  } bits, bitfields;
+  unsigned int u32All;
+};
+
+// General RSRC2 register
+union SPI_SHADER_PGM_RSRC2 {
```

Comment:
Which fields are used actually? Those listed? LDS_SIZE is not always present for each shader stage though previous field definitions, like SCRATCH_EN, USER_SGPR, etc., are identical. Also, the bitwidth of this field is not always 9 bits. GFX10 HS and GS have 10 bits while pre-GFX10 only have 9 bits. Furthermore, the field offset is not the same:)

Reply:
Good point. I'm only using USER_SGPR, so I will remove the other fields.